### PR TITLE
feat: add procedure type

### LINKED ISSUE #556: Add procedure type to `ProcedureInfo`
Run `Deploy` cmd will create a procdure. So we need to distinguish which  procdure is deployment service

### PR DESCRIPTION
**Pull request by dl239**
Run Deploy cmd will create a procdure. So we need to distinguish which procdure is deployment service

### CODE REVIEW

File: hybridse/include/sdk/base.h [hunk 1]
```diff
@@ -129,6 +129,12 @@ class TableSet {
     virtual int32_t Size() { return 0; }
 };
 
+enum ProcedureType {
+    kUnknow = -1,
+    kNormal = 0,
+    kDeployment,
```
Comment (jingchen2222) [inline hybridse/include/sdk/base.h]: Actually, the current Procedure isn't a normal one. It is designed in a special way only to work for `Request` and `BatchRequest` query.
Can we use type like kRequestSp for the current Procedure type. 
And we can add kNormal Procedure in the future which will process procedure in a normal way.

Comment (jingchen2222) [reply hybridse/include/sdk/base.h]: What is the difference between creating a procedure with deploying a procedure?

File: steps/init_env.sh [hunk 2]
```diff
@@ -73,6 +72,10 @@ if [[ ${HYBRIDSE_SOURCE} = "local" ]]; then
     popd
 
 else
+    if [ -d "$THIRDPARTY_PATH/hybridse" ]; then
+        echo "${GREEN}thirdparty/hybridse path: $THIRDPARTY_PATH/hybridse already exist, skip download/install deps${NC}"
+        exit 0
+    fi
```
Comment (jingchen2222) [inline steps/init_env.sh:78]: If so, we had better check hybridse version before.

Comment (dl239) [reply steps/init_env.sh:78]: yes, version check can doing in another issue/pr.

File: hybridse/include/sdk/base.h (hunk 1 above)
Comment (dl239) [reply hybridse/include/sdk/base.h]: we need to distinguish which procdure is a deployment service. If not doing this, we need store deployment like procedure

**Review by vagetablechicken — APPROVED** [review]
LGTM

File: hybridse/include/sdk/base.h [hunk 3]
```diff
@@ -129,6 +129,12 @@ class TableSet {
     virtual int32_t Size() { return 0; }
 };
 
+enum ProcedureType {
```
Comment (tobegit3hub) [inline hybridse/include/sdk/base.h:132]: Can we reuse the ProcedureType from protobuf?

**Review by aceforeverd — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: hybridse/include/sdk/base.h (hunk 3 above)
Comment (tobegit3hub) [reply hybridse/include/sdk/base.h:132]: We may reuse the similar type definition in proto and hybridse later.

Comment (dl239) [reply hybridse/include/sdk/base.h:132]: hybridse cannot use proto defined in src/proto

**Pull request merged**